docs/design/node_agent_config-design.md

```diff
@@ -0,0 +1,517 @@
+					type: string
+					key:
+					description: Key is the taint key that the toleration applies to. Empty means match all taint keys. If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+					type: string
+					operator:
+					description: Operator represents a key's relationship to the value. Valid operators are Exists and Equal. Defaults to Equal. Exists is equivalent to wildcard for value, so that a pod can tolerate all taints of a particular category.
+					type: string
+					tolerationSeconds:
+					description: TolerationSeconds represents the period of time the toleration (which must be of effect NoExecute, otherwise this field is ignored) tolerates the taint. By default, it is not set, which means tolerate the taint forever (do not evict). Zero and negative values will be treated as 0 (evict immediately) by the system.
+					format: int64
+					type: integer
+					value:
+					description: Value is the taint value the toleration matches to. If the operator is Exists, the value should be empty, otherwise just a regular string.
+					type: string
+				type: object
+				type: array
+			type: object
+		supplementalGroups:
+			description: supplementalGroups defines the linux groups to be applied to the Restic Pod
+			items:
+			format: int64
+			type: integer
+			type: array
+		timeout:
+			description: timeout defines the Restic timeout, default value is 1h
+			type: string
+		type: object
+```
+
+### Validations
+
+It is important to warn user that both options `restic` and `nodeagentconfig` are not allowed together.
+
+### Deprecation of the `restic` coiguration option from OADP 1.4+
+
+Since the `restic` configuration option will be deprecated with the OADP 1.4, it is important to inform user to prepare for such deprecation. There were few alternatices considered (see [Alternatives Considered](#alternatives-considered)). We will have three places where the deprecation information will be presented to the user:
+1. Description of the `resic` property will have the deprecation warning
+2. Application log will have `warning` message to inform user that the `restic` will be deprecated if the `restic` was used
+3. Release notes for OADP 1.3 will contain information about new configuration option `nodeagentconfig` that should be used instead of `restic`
+
+
+## Alternatives Considered
+
+- leave `restic` as the only option and do not allow to use `kopia`
+- remove the `restic` and use `kopia` as the default and only available option
+
+### Schema structure
+
+There were alternative schema structure for the `Restic` and new `NodeAgentConfig` considered, however because structures may be directly used outside of the API schema, the decision was to use structure as described in the [Schema structure](#schema-structure) section:
```

Comment:
I am lost in this section. what is the goal of this section? This seems like Alternatives Considered and has a link to itself

Reply:
Ok will rename one.

Reply:
Mistake, should be linked to "New data structures"